keps/262-ConfigurableFailurePolicy/README.md

```diff
@@ -0,0 +1,392 @@
+# KEP-262: Configurable Failure Policy
+
+<!--
+This is the title of your KEP. Keep it short, simple, and descriptive. A good
+title can help communicate what the KEP is and should be considered as part of
+any review.
+-->
+
+<!--
+A table of contents is helpful for quickly jumping to sections of a KEP and for
+highlighting any additional information provided beyond the standard KEP
+template.
+
+Ensure the TOC is wrapped with
+  <code>&lt;!-- toc --&rt;&lt;!-- /toc --&rt;</code>
+tags, and then generate with `hack/update-toc.sh`.
+-->
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [User Stories (Optional)](#user-stories-optional)
+    - [Story 1](#story-1)
+    - [Story 2](#story-2)
+  - [Notes/Constraints/Caveats (Optional)](#notesconstraintscaveats-optional)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [Proposed Failure Policy API](#proposed-failure-policy-api)
+  - [Constraints](#constraints)
+  - [Implementation](#implmentation)
+  - [Test Plan](#test-plan)
+      - [Prerequisite testing updates](#prerequisite-testing-updates)
+    - [Unit Tests](#unit-tests)
+    - [Integration tests](#integration-tests)
+  - [Graduation Criteria](#graduation-criteria)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+<!-- /toc -->
+
+
+## Motivation
+
+JobSet's current FailurePolicy API only allows specifying the max number of JobSet restarts to attempt before marking it as failed.
+
+Users have requested that we should also allow them to configure the failure conditions under which the JobSet will be restarted,
+or when the JobSet should be failed immediately without going through the allowed number of restarts. 
+
+This will allow users to make more efficient use of their computing resources, by avoiding `maxRestarts` number of unnecessary 
+JobSet restarts in the case of a non-retriable failure (e.g., an application code bug), while still restarting in the event of
+retriable failures (e.g., a host maintenace event).
+
+This is especially important for large scale distributed training workloads on expensive hardware accelerators. Recreating the 
+entire JobSet may take ~1 minute or so (per latest scale tests on several node pools of ~3k nodes each), and if `maxRestarts`
+is set to a very high number to allow for disruptions (which occur often at a large scale like this), the workload could
+waste roughly `maxRestarts` number of minutes repeatedly recreating despite the fact that it is doomed to fail due to a non-retriable
```

Comment:
could be more in case the failure doesn't manifest itself immediately

Reply:
Clarified by saying it will '...waste **atleast** maxRestarts number of minutes`